srv/salt/_modules/osd.py

```diff
@@ -379,11 +379,15 @@ def zero_weight(id, wait=True, **kwargs):
     """
     Set weight to zero and wait until PGs are moved
     """
-    settings = {
+    settings = {}
+    storage_keyring = '/etc/ceph/ceph.client.storage.keyring'
+    if os.path.exists(storage_keyring):
+        settings = {
             'keyring': '/etc/ceph/ceph.client.storage.keyring',
             'client': 'client.storage'
-    }
-    settings.update(kwargs)
+        }
+        settings.update(kwargs)
```

Comment:
Update settings with kwargs only if the the ```storage_keyring``` path exists? Not quite obvious why...maybe a comment?